plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -424,6 +450,47 @@ public void testNumericAggregationPushdown()
         }
     }
 
+    @Test
+    public void testCountDistinctWithStringTypes()
+    {
+        if (!(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_INSERT))) {
+            throw new SkipException("Unable to CREATE TABLE to test count distinct");
+        }
+
+        List<String> rows = Stream.of("a", "b", "A", "B", " a ", "a", "b", " b ", "ą")
+                .map(value -> String.format("'%1$s', '%1$s'", value))
+                .collect(toImmutableList());
+        String tableName = "distinct_strings" + randomTableSuffix();
+
+        try (TestTable testTable = new TestTable(getQueryRunner()::execute, tableName, "(t_char CHAR(5), t_varchar VARCHAR(5))", rows)) {
+            if (!(hasBehavior(SUPPORTS_AGGREGATION_PUSHDOWN) && hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY))) {
+                assertQuery("SELECT count(DISTINCT t_varchar) FROM " + testTable.getName(), "VALUES 7");
+                assertQuery("SELECT count(DISTINCT t_char) FROM " + testTable.getName(), "VALUES 7");
+                assertQuery("SELECT count(DISTINCT t_char), count(DISTINCT t_varchar) FROM " + testTable.getName(), "VALUES (7, 7)");
+            }
+            else {
+                assertThat(query("SELECT count(DISTINCT t_varchar) FROM " + testTable.getName()))
+                        .containsAll("VALUES CAST(7 AS BIGINT)")
+                        .isFullyPushedDown();
```

Comment:
in case when `! hasBehavior(SUPPORTS_AGGREGATION_PUSHDOWN_COUNT_DISTINCT)` should this be `isNotFullyPushedDown` after https://github.com/trinodb/trino/pull/8551 ?

cc @hashhar

Reply:
I think yes, see the tests added in https://github.com/trinodb/trino/pull/8551/commits/6473c84d375bf7115571955e7f70dc7725c3431f#diff-6d69520925faa8f876d78d002c7ad94202dfc4a81a9758a5bc3f2ba93b40e023R286-R298

Reply:
I think I do not understand why it should be pushed down if `SUPPORTS_AGGREGATION_PUSHDOWN_COUNT_DISTINCT==true` after #8551

Reply:
A single `count(distinct)` can be pushed down as a `group by`, even for connectors other than PSQL as long as they support agg pushdown, for the group by, and pushdown with varchar equality, so the equality is corrent.